packages/suite-desktop/src/modules/coinjoin.ts

```diff
@@ -51,11 +58,16 @@ export const init: Module = ({ mainWindow }) => {
                     if (method === 'enable') {
                         logger.debug(SERVICE_NAME, `CoinjoinClient enable ${params}`);
                         const response = await client.enable();
-                        // TODO: start coinjoin middleware binary
-                        return response;
+                        try {
+                            await coinjoin.startOnPort(port);
```

Comment:
cool, i would change 2 things:
- throw error, pass it down to caller (suite) to be shown in suite toast
- add `method === 'disable'` case
```
onRequest: async (method, params) => {
        logger.debug(SERVICE_NAME, `CoinjoinClient call ${method}`);
        if (method === 'enable') {
            logger.debug(SERVICE_NAME, `CoinjoinClient binary enable on port ${port}`);
            try {
                await coinjoin.startOnPort(port);
            } catch (err) {
                logger.error(SERVICE_NAME, `Start failed: ${err.message}`);
                throw err; // pass this error to suite toast
            }
        }
        if (method === 'disable') {
            logger.debug(SERVICE_NAME, `CoinjoinClient binary stop`);
            coinjoin.stop();
        }
        // needs type casting
        return (client[method] as any)(...params); // bind method to instance context
    },
```

Reply:
@szymonlesisz Ok
I understand that the code snippet that you pasted is pseudo-code and you didn't want to omit the  `const response = await client.enable();` part.

Reply:
nono, feel free to use my code should be working,

assigning response before calling `coinjoin.startOnPort` is not really necessary, it this fails we dont want to call `client.enable`

`client.enable` should be called when we are sure that everything is good